### PR TITLE
Bugfix/gero 299 hgvs cds hgvs genomic matching misssense to nonsense

### PR DESCRIPTION
**Pull request by dustinbleile**
Add a skipped test for GERO-299 to test_match.py, test_missense_is_not_nonsense.
 - Tests the error.  Error is not yet fixed.

Move GraphKBConnection from __init__.py to util.py.  More standard structure.

Remove the 'verbose' logger, replaced with more standard debug level.

### CODE REVIEW

File: graphkb/util.py [hunk 1]
```diff
@@ -88,3 +81,189 @@ def convert_aa_3to1(three_letter_notation: str) -> str:
...
+
+def join_url(base_url: str, *parts) -> str:
+    """
+    Join parts of a URL into a full URL
+    """
+    if not parts:
+        return base_url
+
+    if base_url.endswith('/'):
+        base_url = base_url[:-1]
```
Comment (mattdoug604) [inline graphkb/util.py:94]: you could replace these two lines with
```python
base_url = base_url.rstrip("/")
```
if you prefer

File: tests/test_match.py [hunk 2]
```diff
@@ -385,7 +385,7 @@ def test_known_fusions(self, conn, known_variant, related_variants):
             assert variant in names
 
     def test_known_fusion_single_gene_no_match(self, conn):
-        known = '(BCR,?):fusion(e.13,e.?)'
+        known = '(TERT,?):fusion(e.1,e.?)'
```
Comment (mattdoug604) [inline tests/test_match.py:388]: Ah ok, I had a question about this in my PR: https://github.com/bcgsc/pori_graphkb_python/pull/66/files#r956280979

I think this is a reasonable change. I just pointed out that it's possible the test will break again depending on what gets uploaded to GraphKB.

Comment (dustinbleile) [reply tests/test_match.py:388]: I agree, but I think we can deal with that if it comes up.

Right now, I think we first just need to get to something that passes these tests, so we can focus on the missense/nonsense category matching test.

File: graphkb/match.py [hunk 3]
```diff
@@ -421,15 +421,15 @@ def match_positional_variant(
...
-        )
+
+    gene1_features = get_equivalent_features(
+        conn,
+        gene1,
+        source=gene_source,
+        is_source_id=gene_is_source_id,
+        ignore_cache=ignore_cache,
     )
+    features = convert_to_rid_list(gene1_features)
```
Comment (mathieulemieux) [inline graphkb/match.py:432]: Just out of curiosity, is nested function calls a bad thing we should avoid or you are planning to need this **gene1_features** variable later on?

File: setup.cfg [hunk 4]
```diff
@@ -10,7 +10,7 @@ include_trailing_comma = true
 [metadata]
 name = graphkb
 url = https://github.com/bcgsc/pori_graphkb_python
-version = 1.6.0
+version = 1.7.0
```
Comment (mathieulemieux) [inline setup.cfg:13]: Shouldn't we merge back the master branch into develop instead?

File: graphkb/util.py [hunk 5]
```diff
@@ -88,3 +81,189 @@ def convert_aa_3to1(three_letter_notation: str) -> str:
 
     result.append(three_letter_notation[last_match_end:])
     return ''.join(result)
+
+
+def join_url(base_url: str, *parts) -> str:
+    """
+    Join parts of a URL into a full URL
+    """
```
Comment (mathieulemieux) [inline graphkb/util.py:89]: On lines 57 and 64, docstrings has been transposed into one-liners. Which way should be prefered?

File: setup.cfg (hunk 4 above)
Comment (dustinbleile) [reply setup.cfg:13]: We cannot merge master into develop as is, because the tests must pass to allow the merge.

File: graphkb/util.py (hunk 5 above)
Comment (dustinbleile) [reply graphkb/util.py:89]: I think the single line is the suggestions of pydocstyle, so I just go with it's suggestions.

File: setup.cfg (hunk 4 above)
Comment (mathieulemieux) [reply setup.cfg:13]: Ok, I see.

**Review by mathieulemieux — APPROVED** [review]

File: graphkb/match.py (hunk 3 above)
Comment (dustinbleile) [reply graphkb/match.py:432]: Nested functions are fine.
I was just comparing gene1_features with gene2_features (below) while debugging, and thought it made the difference between features and secondary_features more clear.

**Pull request merged**